driver-core/src/test/functional/com/mongodb/operation/DropDatabaseOperationSpecification.groovy

```diff
@@ -83,12 +85,20 @@ class DropDatabaseOperationSpecification extends OperationFunctionalSpecificatio
         !databaseNameExists(dbName)
     }
 
-    @IgnoreIf({ !serverVersionAtLeast(3, 4) || serverVersionAtLeast(4, 0) || !isDiscoverableReplicaSet() })
+    @IgnoreIf({ !serverVersionAtLeast(3, 4) || !isDiscoverableReplicaSet() })
     def 'should throw on write concern error'() {
         given:
         getCollectionHelper().insertDocuments(new DocumentCodec(), new Document('documentTo', 'createTheCollection'))
-        assert databaseNameExists(databaseName)
-        def operation = new DropDatabaseOperation(databaseName, new WriteConcern(5))
+
+        // On servers older than 4.0 that don't support this failpoint, use a crazy w value instead
+        def w = serverVersionAtLeast(4, 0) ? 2 : 5
```

Comment:
Minor note: Can we just use 5 and set the failpoint if 4.0+?

Reply:
No, because starting in 4.0 it will report a different error if w is greater than the number of members, before the failpoint is even hit.

Reply:
ACK